hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/ratis/RatisHelper.java

```diff
@@ -101,18 +98,16 @@ public static UUID toDatanodeId(RaftProtos.RaftPeerProto peerId) {
   }
 
   private static String toRaftPeerAddress(DatanodeDetails id, Port.Name port) {
-    if (datanodeUseHostName()) {
-      final String address =
-              id.getHostName() + ":" + id.getPort(port).getValue();
+    final String address;
+    if (Strings.isNullOrEmpty(id.getIpAddress())) {
```

Comment:
We are changing the behavior of this call. Using hostnames if provided should be the preferred way to evaluate the address, not sure why IP address is preferred here (if both are available)

Reply:
IP address is preferred here because that's the default right now, in master. I will change it to hostname.